[FIX] l10n_es_aeat: _parse_aeat_vat_info - show correct aeat identification

Shide opened the pull request:
Conversation from https://github.com/OCA/l10n-spain/pull/2433

- Tests for aeat_identification_type and aeat_identification
- Adds to ormcache those fields
- If user has aeat_identification_type, use aeat_identification

MT-706 @moduon

@pedrobaeza

pedrobaeza commented on l10n_es_aeat/models/res_partner.py:
```diff
@@ -72,8 +75,8 @@ def _parse_aeat_vat_info(self):
+        if self.aeat_identification_type:
+            identifier_type = self.aeat_identification_type
+            vat_number = self.aeat_identification
+
```
Evita por favor líneas vacías en medio del método.

Shide committed:
[FIX] l10n_es_aeat: _parse_aeat_vat_info - show correct aeat identification

- Tests for aeat_identification_type and aeat_identification
- Adds to ormcache those fields
- If user has aeat_identification_type, use aeat_identificationç

[FIX] l10n_es_aeat: Fix test_parse_vat_info_aeat_identification tests

pedrobaeza commented on l10n_es_aeat/tests/test_l10n_es_aeat.py:
```diff
@@ -35,6 +35,32 @@ def test_parse_vat_info_es_w_prefix(self):
+        self.assertEqual(country_code, "ES")
+        self.assertEqual(identifier_type, "03")
+        self.assertEqual(vat_number, "MY_PASSPORT_03")
+
```
No te puse el mismo comentario por no cansar, pero es lo mismo aquí (y en la siguiente) sobre no introducir líneas vacías en medio del método. El propio comentario con el highlighting ya actúa de separador. Igual que prevenimos líneas largas para evitar tener scroll lateral, tenemos lo mismo con las líneas vacías para el scroll vertical.

Shide replied:
Listo!

pedrobaeza commented:
/ocabot merge patch

pedrobaeza commented:
@Shide vas a añadir en otro PR los tipos de identificación faltantes, ¿verdad?

Shide commented:
No lo tengo planeado. De momento todo funciona bien. No obstante si decidimos subirlos, aún tardaremos un poquillo.

pedrobaeza commented:
Bueno, el incluir esos tipos responde a los puntos que expuse en https://github.com/OCA/l10n-spain/issues/2247#issuecomment-1189909704 para la incidencia que planteáis, pero bueno, puedo yo añadirlos si no lo vais a hacer vosotros.
